app/src/main/java/com/hedvig/app/SplashActivity.kt

```diff
@@ -47,88 +52,7 @@ class SplashActivity : BaseActivity(R.layout.activity_splash) {
 
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
-        handleFirebaseDynamicLink(intent, null)
```

Comment:
Before, we were using the intent coming from `onNewIntent` to pass to `FirebaseDynamicLinks.getInstance().getDynamicLink(intent)` . Now we are using `Activity.getIntent()` instead since we don't pass it from here, but the documentation for it says "Return the intent that started this activity." but in the case where we get the deep link from onNewIntent() the one that started the activity wouldn't be the same as the one passed inside the onNewIntent() right?

Would this be a bug in case the activity is already open get get a new intent with a deep link, or does the Intent returned from `Activity.getIntent()` get updated after the `onNewIntent` happens? The docs on `onNewIntent` seem to mention that "Note that getIntent still returns the original Intent." so I just looking at the code makes me think this would make these links not work if the activity is already open 🤔 What do you think? @hugokallstrom

Reply:
I see! I think this was an issue in the original code as well though. 

`handleFirebaseDynamicLink(intent, null)` this will call the function with loginStatus as null, so the checks in `handleReferralsDeepLink()`, `handleDirectDebitDeepLink()` and `handleForeverDeepLink()` will call `startDefaultActivity()` with null loginStatus as parameter. This will finally call `handleFirebaseDynamicLink(intent, loginStatus)` in `navigateToActivity()`. A bit hard to follow but I think thats whats going on 😅

Reply:
Yeah, we don't want it to pass null all the time either.
How about then having `getLoginStatusAndNavigate` and `navigateToActivity` receive a nullable intent parameter? That way if the deep link comes from the onNewIntent it can use the correct one, and if it doesn't we can do like
```kotlin
val intent = intentComingFromParamters ?: this@Activity.intent
dynamicLinkHandler.handleIntent(intent, loginStatus)
```
so that we actually use the correct intent in this case.

Reply:
Good idea, fixed 👍